examples/passport-login/src/application.ts

```diff
@@ -35,6 +39,19 @@ export class OAuth2LoginApplication extends BootMixin(
     this.component(AuthenticationComponent);
     this.component(CrudRestComponent);
 
+    this.bind('passport-init-mw').to(toInterceptor(passport.initialize()));
+    this.bind('passport-session-mw').to(toInterceptor(passport.session()));
+    passport.serializeUser(function(user: any, done) {
+      done(null, user);
+    });
+    
+    passport.deserializeUser(function(user: any, done) {
+      done(null, user);
+    });
+    this.bind('passport-facebook').toProvider(FacebookOauth);
+    this.bind('passport-google').toProvider(GoogleOauth);
+    this.bind('passport-oauth2').toProvider(CustomOauth2);
+
```

Comment:
You can compose 5 interceptors into one (depending on https://github.com/strongloop/loopback-next/pull/5343)

```ts
import {composeInterceptors} from '@loopback/context';
const authInterceptor = composeInterceptors(
  'passport-init-mw', // or use `toInterceptor(passport.initialize())` directly without binding it
  'passport-session-mw', // or use `toInterceptor(passport.session())` 
  'passport-facebook', 
  'passport-google', 
  'passport-oauth2');

// ...
@intercept(authInterceptor)

```

Reply:
perfect, thank you